frontend/src/components/dashboard/Analysis.tsx

```diff
@@ -0,0 +1,432 @@
+  setXAxis: (axis: AxesCriterionType) => void;
+  yAxis: AxesCriterionType;
+  setYAxis: (axis: AxesCriterionType) => void;
+  taskType: TaskType;
+  categories: CategoryWithGroups[];
+  groups: Group[];
+  splits: ChartSplit[];
+  setSplits: (
+    updateSplits: (currentSplits: ChartSplit[]) => ChartSplit[],
+  ) => void;
+}) => {
+  const intl = useIntl();
+
+  const [tooltipDataPoints, setTooltipDataPoints] = useState<
+    PointWithAdditionalData[]
+  >([]);
+  const tooltipRef = useRef<HTMLDivElement>(null);
+
+  const xAxisConfig = useMemo(() => getAxisConfig(xAxis), [xAxis]);
+  const yAxisConfig = useMemo(() => getAxisConfig(yAxis), [yAxis]);
+
+  const chartRef = useRef<ChartJsChart | null>(null);
+
+  const chartData = useMemo<ChartData<"bubble">>(
+    () => ({
+      datasets: categories.map((category) => ({
+        label: getCategoryName(intl, category.category),
+        data: category.solutions.map((solution) => ({
+          x: solution.xAxisValue,
+          y: solution.yAxisValue,
+          r: 10,
+          additionalData: {
+            groupName: solution.groupKey,
+            studentPseudonym: solution.studentPseudonym,
+          } as AdditionalChartData,
+        })),
+
+        backgroundColor: getCanvasPattern(category.category),
+      })),
+    }),
+    [categories, intl],
+  );
+
+  const onTooltip = useCallback(
+    ({
+      chart,
+      tooltip,
+    }: {
+      chart: ChartJsChart;
+      tooltip: TooltipModel<"bubble">;
+    }) => {
+      const tooltipElement = tooltipRef.current;
+
+      if (!tooltipElement) {
+        return;
+      }
+
+      if (tooltip.opacity === 0) {
+        // Hide the tooltip
+        setTooltipDataPoints([]);
+        return;
+      }
+
+      const { offsetLeft: positionX, offsetTop: positionY } = chart.canvas;
+
+      // update position of tooltip
+      tooltipElement.style.left = positionX + tooltip.caretX + "px";
+      tooltipElement.style.top = positionY + tooltip.caretY + "px";
+
+      setTooltipDataPoints(
+        tooltip.dataPoints.map(
+          (dataPoint) =>
+            chart.data.datasets[dataPoint.datasetIndex].data[
+              dataPoint.dataIndex
+            ] as BubbleDataPoint & {
+              additionalData: AdditionalChartData;
+            },
+        ),
+      );
+    },
+    [],
+  );
+
+  const annotations = useMemo<_DeepPartialObject<AnnotationPluginOptions>>(
+    () => ({
+      // allow annotations in axes
+      clip: false,
+
+      annotations: [
+        ...splits.flatMap<AnnotationOptions>((split) => {
+          const onClickLabel = (ctx: EventContext, evt: ChartEvent) => {
+            if (
+              (evt.native && "handled" in evt.native) ||
+              "blockDeletes" in ctx.chart
+            ) {
+              return;
+            }
+
+            // @ts-expect-error This is an annoying way working around the fact that stopImmediatePropagation doesn't work
+            evt.native["handled"] = true;
```

Comment:
Since we have this horrible hack that necessarily shows up in two places, can we abstract it in a utility function documenting the issue ? Perhaps something like `isAlreadyHandled(event) / markAsHandled(event)` ?